### ISSUE TITLE
MA0042 is also reported for non-async methods

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#538]
### Version of the Meziantou.Analyzer NuGet package

2.0.54

### Rule Identifier

MA0042

### Target Framework

net7.0

### C# Language version

C# 11

### Description

The doc page for **MA0042** makes me believe that it will not report a diagnostic for a method without **async** modifier:

> The rule only reports a diagnostic when the calling method is already async.

However, **MA0042** is reported when you have a using statement and the instance implements **IAsyncDisposable**.

### Reproduction Steps

Minimal code:

```c#
public static Task RunNotAsync(Resource resource)
{
    // This results in MA0042: Prefer using 'await using'
    using (resource)
    {
    }

    return Task.CompletedTask;
}

public sealed class Resource : IDisposable, IAsyncDisposable
{
    public void Dispose()
    {
    }

    public ValueTask DisposeAsync()
    {
        return ValueTask.CompletedTask;
    }
}
```

**Comment by meziantou** [comment]
MA0042 is reported when the enclosing method has the `async` keyword or when the return type is awaitable (`Task`, `ValueTask`, `IAsyncEnumerable`, etc.). In this case, you can use the `await` keyword without changing the signature (adding `async` is an implementation detail, not a breaking change). So, in your example, I think the MA0042 should be reported as you can add the `async` keyword without changing the signature.

Maybe the documentation should be clarified.

Side-note: I recently changed the rule to detect awaitable types instead of hardcoding `Task` and `ValueTask`. But, I think I made a mistake. Instead of detecting awaitable types, it should detect types that support the `async` keyword (`Task`, `Task<T>`, `IAsyncEnumerable<T>`, `IAsyncEnumerator<T>`, types decorated with `[AsyncMethodBuilder]`)
